ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts

```diff
@@ -0,0 +1,216 @@
+import { TranslateService } from '@ngx-translate/core';
+import {SettingsService} from '@ofServices/settings.service';
+import {Subject} from 'rxjs';
+
+
+class Message {
+    text: string;
+    display: boolean;
+}
+
+function getI18nPrefixFromProcess(process: Process): string {
+    return process.id + '.' + process.version + '.';
+}
+
+@Component({
+    selector: 'of-feedconfiguration',
+    templateUrl: './feedconfiguration.component.html',
+    styleUrls: ['./feedconfiguration.component.scss']
+})
+
+export class FeedconfigurationComponent implements OnInit, OnDestroy {
+    feedConfigurationForm: FormGroup;
+
+    processesDefinition: Process[];
+    processGroups: {idGroup: string, processes: string[]}[];
+    processesWithoutGroup: string[];
+    currentUserWithPerimeters: UserWithPerimeters;
+    processesStatesLabels: Map<string, { processLabel: string,
+                                           states:
+                                               { stateLabel: string,
+                                                 stateControlIndex: number
+                                               }[]
+                                         }>;
+    preparedListOfProcessesStates: { processId: string,
+                                     stateId: string }[];
+
+    modalRef: NgbModalRef;
+
+    public displaySendResultOk = false;
+    public displaySendResultError = false;
+    messageAfterSavingSettings: string;
+
+    unsubscribe$: Subject<void> = new Subject<void>();
+
+    constructor(private formBuilder: FormBuilder,
+                private store: Store<AppState>,
+                private userService: UserService,
+                private processesService: ProcessesService,
+                private modalService: NgbModal,
+                private translateService: TranslateService,
+                private settingsService: SettingsService,
+    ) {
+        this.processesDefinition = this.processesService.getAllProcesses();
+        this.processGroups = this.processesService.getProcessGroups();
+        this.currentUserWithPerimeters = this.userService.getCurrentUserWithPerimeters();
+        this.feedConfigurationForm = this.formBuilder.group({
+            processesStates: new FormArray([])
+        });
+        this.processesStatesLabels = new Map<string, {processLabel: string,
+                                                        states:
+                                                            { stateLabel: string,
+                                                              stateControlIndex: number
+                                                            }[]
+                                                       }> ();
+        this.preparedListOfProcessesStates = [];
+        this.processesWithoutGroup = [];
+
+        this.computePreparedListOfProcessesStatesAndProcessesStatesLabels();
+        this.makeProcessesWithoutGroup();
+        this.addCheckboxesInFormArray();
```

Comment:
this.addCheckboxesInFormArray() should be call i guess in the ngOnInit method . There is actually a bug when you change screen and return after to this screen , the checkbox selected and saved are not checked anymore

Reply:
Done.